bitnami/janusgraph/templates/deployment.yaml

```diff
@@ -0,0 +1,293 @@
+          {{- else if ne .Values.resourcesPreset "none" }}
+          resources: {{- include "common.resources.preset" (dict "type" .Values.resourcesPreset) | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: empty-dir
+              mountPath: /ext-dir
+              subPath: ext-dir
+            - name: empty-dir
+              mountPath: /base-dir
+              subPath: base-dir-extra
+        {{- if and .Values.volumePermissions.enabled .Values.persistence.enabled }}
+        # Adjust volume permissions
+        {{- include "janusgraph.volumePermissionsInitContainer" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.initContainers }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      containers:
+        - name: janusgraph
+          image: {{ include "janusgraph.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: BITNAMI_DEBUG
+              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+            - name: JANUSGRAPH_PORT_NUMBER
+              value: {{ .Values.containerPorts.gremlin | quote }}
+            - name: JAVA_OPTIONS
+              value: {{ include "janusgraph.javaOpts" . | quote }}
+            - name: JANUSGRAPH_JMX_METRICS_ENABLED
+              value: {{ ternary "true" "false" .Values.metrics.enabled | quote }}
+            - name: GREMLIN_AUTOCONFIGURE_POOL
+              value: "true"
+            {{- if (include "janusgraph.storage.username" .)}}
+            - name: JANUSGRAPH_CFG_STORAGE_USERNAME
+              value: {{ include "janusgraph.storage.username" . | quote }}
+            - name: JANUSGRAPH_CFG_STORAGE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "janusgraph.storage.password.secretName" . }}
+                  key: {{ include "janusgraph.storage.password.secretKey" . }}
+            {{- end }}
```

Comment:
Understood, although it may require a new release of the container image, since the `JANUSGRAPH_CFG_STORAGE_PASSWORD` has the same behavior as other `APPNAME_CFG_` variables where they are directly mapped to the configuration file.